client-ui/src/app/model/property.ts

```diff
@@ -41,3 +41,7 @@ export const enum PropertyType {
     Properties = "Properties",
     Error = "Error"
 }
+
+export function PropertyTypeAware(constructor: Function) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'PropertyTypeAware' to have a typedef](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4605551)